src/Raven.Client/Util/StreamWithTimeout.cs

```diff
@@ -195,9 +222,25 @@ public override Task WriteAsync(byte[] buffer, int offset, int count, Cancellati
 
         private Task WriteAsyncWithTimeout(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            _writeCts?.Dispose();
-            _writeCts = cancellationToken == default ? new CancellationTokenSource() : CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
-            _writeCts.CancelAfter(_writeTimeout);
+            if (_writeCts == null)
+            {
+                _writeCts = cancellationToken == default ? new CancellationTokenSource() : CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                _writeSw = Stopwatch.StartNew();
```

Comment:
you are not calling CancelAfter in this flow

Reply:
done